### PR TITLE
refactor: pass callback url from app as per env

### PR DESCRIPTION
**Pull request by Nisha1306**

### CODE REVIEW

File: src/api/vipps-login/index.ts [hunk 1]
```diff
@@ -13,14 +13,14 @@ export default (server: Hapi.Server) => () => {
         projectId: PROJECT_ID
     })
 
-    const getClient = async () => {
+    const getClient = async (callbackUrl: string) => {
         const iss = await Issuer.discover(VIPPS_BASE_URL + '/access-management-1.0/access/');
```
Comment (torfjor) [inline src/api/vipps-login/index.ts:17]: While we're here, maybe we should cache the `Issuer` in an outer scope and return the cached copy if has a value? Waiting for OIDC discovery for each authorization request is wasteful. The JWKS will be kept and automatically refreshed based on the expiry in the response.

**Review by torfjor — APPROVED** [review]
One small suggestion for performance / resource optimization. 

Otherwise 👍

File: src/api/vipps-login/index.ts (hunk 1 above)
Comment (Nisha1306) [reply src/api/vipps-login/index.ts]: how about now?

File: src/api/vipps-login/index.ts [hunk 2]
```diff
@@ -13,14 +13,14 @@ export default (server: Hapi.Server) => () => {
         projectId: PROJECT_ID
     })
 
-    const getClient = async () => {
-        const iss = await Issuer.discover(VIPPS_BASE_URL + '/access-management-1.0/access/');
+    const getClient = async (callbackUrl: string) => {
+        const iss = await getOpenIdClientIssuer()
         return new iss.Client({
             client_id: VIPPS_CLIENT_ID || '',
             client_secret: VIPPS_CLIENT_SECRET || '',
```
Comment (gorandalum) [inline src/api/vipps-login/index.ts:20]: Verify these env variables at startup.

**Review by gorandalum — APPROVED** [review]
Looks good. Just verify some env parameters at app startup.

**Pull request merged**